client/rest/src/test/java/org/elasticsearch/client/RestClientIntegTests.java

```diff
@@ -229,6 +229,18 @@ public void testPathPrefix() throws IOException {
         }
     }
 
+    public void testPath() throws IOException {
+        Response esResponse;
+        for (final String method : getHttpMethods()) {
+            try {
+                esResponse = restClient.performRequest(method, null);
```

Comment:
this is not the intended way to use this method. Why should `null` be accepted? I would rather throw NPE earlier than accept it and convert it to something else.